src/Components/Alert/AlertProvider.tsx

```diff
@@ -0,0 +1,208 @@
+    preview: null,
+    visible: false,
+  }
+  const [current, setCurrent] = useState<
+    "ALERT_DETAILS" | "ALERT_FILTERS" | "ALERT_CONFIRMATION"
+  >("ALERT_DETAILS")
+
+  useEffect(() => {
+    const criteria =
+      getAllowedSearchCriteria(initialCriteria as SearchCriteriaAttributes) ??
+      {}
+
+    dispatch({ type: "SET_CRITERIA", payload: criteria })
+  }, [initialCriteria])
+
+  const handleComplete = async () => {
+    try {
+      const reponse = await submitMutation({
+        variables: {
+          input: {
+            attributes: state.criteria,
+            userAlertSettings: state.settings,
+          },
+        },
+      })
+      const searchCriteriaID =
+        reponse.createSavedSearch?.savedSearchOrErrors.internalID
+
+      if (searchCriteriaID) {
+        dispatch({
+          type: "SET_SEARCH_CRITERIA_ID",
+          payload: searchCriteriaID,
+        })
+
+        createdAlert(searchCriteriaID)
+      }
+      setCurrent("ALERT_CONFIRMATION")
+    } catch (error) {
+      console.error("Alert/useAlertContext", error)
+    }
+  }
+
+  const onReset = (): State => {
+    setCurrent("ALERT_DETAILS")
+
+    return initialState
+  }
+
+  const onShow = (state: State): State => {
+    if (!isLoggedIn) {
+      showAuthDialog({
+        mode: "SignUp",
+        options: {
+          title: "Sign up to create your alert",
+          afterAuthAction: {
+            action: Intent.createAlert,
+          },
+        },
+        analytics: {
+          intent: Intent.createAlert,
+          contextModule: ContextModule.createAlert,
+        },
+      })
+
+      return state
+    }
+
+    return {
+      ...state,
+      visible: true,
+    }
+  }
+
+  const [state, dispatch] = useReducer(reducer(onShow, onReset), initialState)
+
+  // Fetching the display name and labels for alert name input placeholder and the criteria pills, when the criteria changes.
+
+  const { debouncedValue: debouncedCriteria } = useDebouncedValue({
+    value: state.criteria,
+    delay: 200,
+    settings: { leading: true },
+  })
+
+  useEffect(() => {
+    const subscription = fetchQuery<AlertProviderPreviewQuery>(
+      relayEnvironment as Environment,
+      graphql`
+        query AlertProviderPreviewQuery(
+          $attributes: PreviewSavedSearchAttributes
+        ) {
+          viewer {
+            previewSavedSearch(attributes: $attributes) {
+              displayName
+              labels {
+                displayValue
+                field
+                value
+              }
+            }
+          }
+        }
+      `,
+      { attributes: debouncedCriteria as PreviewSavedSearchAttributes }
+    ).subscribe({
+      next: data => {
+        if (!data?.viewer?.previewSavedSearch) return
+
+        dispatch({
+          type: "SET_PREVIEW",
+          payload: data?.viewer?.previewSavedSearch as PreviewSavedSearch,
+        })
+      },
+    })
+
+    return () => {
+      subscription.unsubscribe()
```

Comment:
@olerichter00 Hopefully this reduces the chance of any runaway network requests causing trouble after the component dismounts.